Error: unused parameter 'ConfCxt' with `-DWASMEDGE_BUILD_AOT_RUNTIME=OFF`

When building with `cmake -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_BUILD_AOT_RUNTIME=OFF ..` (as instructed in `doc/build.md`), I got the following error message:
```
[ 98%] Building CXX object lib/api/CMakeFiles/wasmedge_c.dir/wasmedge.cpp.o
/root/WasmEdge/lib/api/wasmedge.cpp: In function 'WasmEdge_CompilerContext* WasmEdge_CompilerCreate(const WasmEdge_ConfigureContext*)':
/root/WasmEdge/lib/api/wasmedge.cpp:754:58: error: unused parameter 'ConfCxt' [-Werror=unused-parameter]
  754 | WasmEdge_CompilerCreate(const WasmEdge_ConfigureContext *ConfCxt) {
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/root/WasmEdge/lib/api/wasmedge.cpp: In function 'WasmEdge_Result WasmEdge_CompilerCompile(WasmEdge_CompilerContext*, const char*, const char*)':
/root/WasmEdge/lib/api/wasmedge.cpp:767:31: error: unused parameter 'Cxt' [-Werror=unused-parameter]
  767 |     WasmEdge_CompilerContext *Cxt, const char *InPath, const char *OutPath) {
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
/root/WasmEdge/lib/api/wasmedge.cpp:767:48: error: unused parameter 'InPath' [-Werror=unused-parameter]
  767 |     WasmEdge_CompilerContext *Cxt, const char *InPath, const char *OutPath) {
      |                                    ~~~~~~~~~~~~^~~~~~
/root/WasmEdge/lib/api/wasmedge.cpp:767:68: error: unused parameter 'OutPath' [-Werror=unused-parameter]
  767 |     WasmEdge_CompilerContext *Cxt, const char *InPath, const char *OutPath) {
      |                                                        ~~~~~~~~~~~~^~~~~~~
cc1plus: all warnings being treated as errors
```

### Versions
- WasgmEdge: `master (c4452d778636d3af207a0602dbec6276b8099847)`
- Docker image: `wasmedge/wasmedge:ubuntu-build-gcc (04cd593f0f1e)`

--- a/lib/api/wasmedge.cpp
+++ b/lib/api/wasmedge.cpp
@@ -751,7 +751,8 @@ WasmEdge_ASTModuleDelete(WasmEdge_ASTModuleContext *Cxt) {
 /// >>>>>>>> WasmEdge AOT compiler functions >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
 
 WASMEDGE_CAPI_EXPORT WasmEdge_CompilerContext *
-WasmEdge_CompilerCreate(const WasmEdge_ConfigureContext *ConfCxt) {
+WasmEdge_CompilerCreate(const WasmEdge_ConfigureContext *ConfCxt
+                        [[maybe_unused]]) {
 #ifdef WASMEDGE_BUILD_AOT_RUNTIME
   if (ConfCxt) {
     return new WasmEdge_CompilerContext(ConfCxt->Conf);
@@ -764,7 +765,8 @@ WasmEdge_CompilerCreate(const WasmEdge_ConfigureContext *ConfCxt) {
 }
 
 WASMEDGE_CAPI_EXPORT WasmEdge_Result WasmEdge_CompilerCompile(
-    WasmEdge_CompilerContext *Cxt, const char *InPath, const char *OutPath) {
+    WasmEdge_CompilerContext *Cxt [[maybe_unused]],
+    const char *InPath [[maybe_unused]], const char *OutPath [[maybe_unused]]) {
 #ifdef WASMEDGE_BUILD_AOT_RUNTIME
   return wrap(
       [&]() -> WasmEdge::Expect<void> {

